stingray/modeling/gpmodeling.py

```diff
@@ -0,0 +1,850 @@
+
+        Parameters
+        ----------
+        name : str
+            Name of the parameter.
+            Should be from the names of the parameter list used or from the names of parameters
+            used in the prior_function
+
+        n : int, default 0
+            The index of the parameter to be plotted.
+            For multivariate parameters, the index of the specific parameter to be plotted.
+
+        key: jax.random.PRNGKey, default ``random.PRNGKey(1234)``
+            Random key for the weighted sampling
+
+        axis : list, tuple, string, default ``None``
+            Parameter to set axis properties of ``matplotlib`` figure. For example
+            it can be a list like ``[xmin, xmax, ymin, ymax]`` or any other
+            acceptable argument for ``matplotlib.pyplot.axis()`` method.
+
+        save : bool, optionalm, default ``False``
+            If ``True``, save the figure with specified filename.
+
+        filename : str
+            File name and path of the image to save. Depends on the boolean ``save``.
+
+        Returns
+        -------
+        plt : ``matplotlib.pyplot`` object
+            Reference to plot, call ``show()`` to display it
+        """
+        if rkey is None:
+            rkey = random.PRNGKey(1234)
+
+        nsamples = self.Results.total_num_samples
+        log_p = self.Results.log_dp_mean
+        samples = self.Results.samples[name].reshape((nsamples, -1))[:, n]
+
+        weights = jnp.where(jnp.isfinite(samples), jnp.exp(log_p), 0.0)
+        log_weights = jnp.where(jnp.isfinite(samples), log_p, -jnp.inf)
+        samples_resampled = resample(
+            rkey, samples, log_weights, S=max(10, int(self.Results.ESS)), replace=True
+        )
+
+        nbins = max(10, int(jnp.sqrt(self.Results.ESS)) + 1)
+        binsx = jnp.linspace(*jnp.percentile(samples_resampled, jnp.asarray([0, 100])), 2 * nbins)
+
+        plt.hist(
+            np.asarray(samples_resampled),
+            bins=binsx,
+            density=True,
+            alpha=1.0,
+            label=name,
+            fc="None",
+            edgecolor="black",
+        )
+        sample_mean = jnp.average(samples, weights=weights)
+        sample_std = jnp.sqrt(jnp.average((samples - sample_mean) ** 2, weights=weights))
+        plt.axvline(sample_mean, color="red", linestyle="dashed", label="mean")
+        plt.axvline(sample_mean + sample_std, color="green", linestyle="dotted")
+        plt.axvline(sample_mean - sample_std, linestyle="dotted", color="green")
+        plt.title("Weighted Posterior Histogram of " + str(name))
+        plt.xlabel(name)
+        plt.ylabel("Probability Density")
+        plt.legend()
+        if axis is not None:
+            plt.axis(axis)
+
+        if save:
+            if filename is None:
+                plt.savefig(str(name) + "_Weighted_Posterior_plot.png")
+            else:
+                plt.savefig(filename)
+        return plt
+
+    def corner_plot(
```

Comment:
I'm not sure we need our own implementation of a corner plot. I think we can just use the `corner.py` package here?

Reply:
Outdated now